bundles/admin/admin-hierarchical-layerlist/instance.js

```diff
@@ -0,0 +1,364 @@
+
+            var ajaxData = {
+                orders: [],
+                parent: (parentNode.a_attr && parentNode.a_attr['data-group-id']) ? parentNode.a_attr['data-group-id'] : -1
+            };
+
+            parentNode.children.forEach(function(nodeId) {
+                var node = data.data.origin.get_node(nodeId);
+                var order = {
+                    type: node.type, // groups, subgroup, subgroups
+                    id: (node.type === 'layer') ? node.a_attr['data-layer-id'] : node.a_attr['data-group-id']
+                };
+                if (ajaxData.parent != -1 && ajaxData.parent != draggedNodeOldParentId && draggedNode.type === node.type && order.id === draggedId) {
+                    order.oldParent = draggedNodeOldParentId;
+                }
+                ajaxData.orders.push(order);
+            });
+
+
+            jQuery.ajax({
+                type: 'POST',
+                contentType: 'application/json; charset=UTF-8',
+                url: me.sandbox.getAjaxUrl('LayerAndGroupOrder'),
+                data: JSON.stringify(ajaxData),
+                error: function(response) {
+                    var errorDialog = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+                    errorDialog.show(me.locale('errors.nodeDropSave.title'), me.locale('errors.nodeDropSave.message'));
+                    errorDialog.fadeout();
+                    me.service.trigger('jstree-order-changed', {
+                        ajax: true,
+                        success: false
+                    });
+                },
+                success: function(response) {
+                    var successDialog = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+                    successDialog.show(me.locale('succeeses.nodeDropSave.title'), me.locale('succeeses.nodeDropSave.message'));
+                    successDialog.fadeout();
+                    me.service.trigger('jstree-order-changed', {
+                        ajax: true,
+                        success: true
+                    });
+                }
+            });
+            me.service.trigger('jstree-order-changed', data);
+        },
+
+
+        /*******************************************************************************************************************************
+        /* PUBLIC METHODS
+        *******************************************************************************************************************************/
+        getLocalization: function(key) {
+            if (!this.locale) {
+                this.locale = Oskari.getMsg.bind(null, this.getName());
+            }
+            if (key) {
+                return this.locale(key);
+            }
+            return this.locale;
+        },
+        getName: function() {
+            return "AdminHierarchicalLayerList";
+        },
+        start: function() {
+            var me = this;
+            if (!me.service) {
```

Comment:
Add a comment here like "Only works with hierarchical layer selector, if the service it provides is not available we cannot proceed".